src/MongoDB.Driver.Core/Core/Connections/ClientDocumentHelper.cs

```diff
@@ -59,11 +59,22 @@ internal static BsonDocument CreateDriverDocument()
 
         internal static BsonDocument CreateDriverDocument(string driverVersion)
         {
+            var driverName = "mongo-csharp-driver";
+            if (IsLegacyLoaded())
+            {
+                driverName = $"{driverName}|legacy";
+            }
+
             return new BsonDocument
             {
-                { "name", "mongo-csharp-driver" },
+                { "name", driverName },
                 { "version", driverVersion }
             };
+
+            bool IsLegacyLoaded()
+            {
+                return Type.GetType("MongoDB.Driver.MongoServer, MongoDB.Driver.Legacy") != null;
```

Comment:
it's unfortunate that AppDomain is not available on standard1.5 :(

Reply:
FYI: maybe it will be interesting to read about suggestions how we can use local functions: http://dontcodetired.com/blog/post/Refactoring-Code-to-Use-C-Local-Functions